src/utils/experiment-tracking-utils.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Take a value and return a meaningful value for display on experiment tracking tables
+ * @param {value} value The value to be sanitized
+ * @returns A sanitized value
+ */
+export const sanitizeValue = (value) => {
+  if (value === '' || value === null || value === undefined) {
+    return '-';
+  } else if (typeof value === 'object' || typeof value === 'boolean') {
+    return JSON.stringify(value);
+  }
+
+  return value;
+};
+
+/**
+ * Takes a set of run metadata and run tracking data to construct the array object for csv export
+ * @param {array} runMetadata The set of runMetadata
+ * @param {array} runTrackingData The set of runTrackingData
+ * @returns An array formatted for CSV export
+ */
+export const constructExportData = (runMetadata, runTrackingData) => {
+  let csvData = [];
+  if (runMetadata && runTrackingData) {
+    // obtain runMetadata
+    const runTitle = runMetadata?.map((run) => sanitizeValue(run.title));
+    const createdBy = runMetadata?.map((run) => sanitizeValue(run.author));
+    const gitSha = runMetadata?.map((run) => sanitizeValue(run.gitSha));
+    const gitBranch = runMetadata?.map((run) => sanitizeValue(run.gitBranch));
+    const runCommand = runMetadata?.map((run) => sanitizeValue(run.runCommand));
+    const notes = runMetadata?.map((run) => sanitizeValue(run.notes));
+
+    csvData.push(
+      ['Title', ...runTitle],
+      ['Created By', ...createdBy],
+      ['Git SHA', ...gitSha],
+      ['Git Branch', ...gitBranch],
+      ['Run Command', ...runCommand],
+      ['Notes', ...notes]
+    );
+
+    // create empty line between metadata fields and tracking data fields
+    csvData.push([]);
+
+    runTrackingData.forEach((trackingDataset) => {
+      const { datasetName, data } = trackingDataset;
+      csvData.push([datasetName]);
+      const dataKeyNames = Object.keys(data);
+      dataKeyNames.forEach((key) => {
+        let keyData = [key];
+        data[key].forEach((datafield) => keyData.push(datafield.value));
+        csvData.push(keyData);
+      });
+      csvData.push([]);
+    });
+  }
+  return csvData;
+};
+
+/**
+ * Take a the runMetadata list to generate a meaningful file name for csv export
+ * @param {array} runMetadata The set of runMetadata to be exported
+ * @returns A string to be used as the file name
+ */
+export const generateCsvFileName = (runMetadata) => {
```

Comment:
This products some pretty crazy file names, such as this: 

```
rundata-2021-12-16T09.46.21.973Z-2021-12-16T09.47.19.553Z-2021-12-16T09.48.12.714Z.csv
```

I don't have a better suggestion for what it should be though. I just find the above really hard to parse. Have you asked Yetu/design if they have any guidance?

Reply:
yea indeed I do share the same thoughts with you about the file name - we haven't discussed it yet and there isn't any particular requirement for this, I thought it would be great to include the run id in the file name which, given the unique structure of our run id, leads to this 'very readable' file name 😆 

The other way is to simply name the file as `rundata.csv` without including the run ids - we can see if anyone complains about this current file name arrangement and revert if needed?

Reply:
I like your `rundata.csv` idea (should it be `runData.csv` or even `run-data.csv`?). Maybe we aim for simplicity first and then if users want more specificity we iterate.

Reply:
sounds good, `run-data.csv` it is then 😄